rllib/examples/eager_execution.py

```diff
@@ -103,22 +104,26 @@ def get_default_policy_class(self, config):
     args = parser.parse_args()
     ModelCatalog.register_custom_model("eager_model", EagerModel)
 
-    config = {
-        "env": "CartPole-v1",
+    config = (
+        AlgorithmConfig()
+        .environment("CartPole-v1")
+        .framework("tf2")
+        .rollouts(num_rollout_workers=0)
+        .training(model={"custom_model": "eager_model"})
         # Use GPUs iff `RLLIB_NUM_GPUS` env var set to > 0.
-        "num_gpus": int(os.environ.get("RLLIB_NUM_GPUS", "0")),
-        "num_workers": 0,
-        "model": {"custom_model": "eager_model"},
-        "framework": "tf2",
-    }
+        .resources(num_gpus=int(os.environ.get("RLLIB_NUM_GPUS", "0")))
```

Comment:
I think, since we have config objects now, we should default to num_gpus=None and check for RLLIB_NUM_GPUS when freezing the config object. This will make this not very pretty and super redudant line unnecessary.